include/zephyr/drivers/gpio/gpio_utils.h

```diff
@@ -17,14 +17,16 @@
 #include <zephyr/sys/__assert.h>
 #include <zephyr/sys/slist.h>
 
+/* Please use GPIO_DT_INST_PORT_PIN_MASK_NGPIOS_EXC et al instead */
 #define GPIO_PORT_PIN_MASK_FROM_NGPIOS(ngpios)			\
+	__DEPRECATED_MACRO					\
 	((gpio_port_pins_t)(((uint64_t)1 << (ngpios)) - 1U))
 
 #define GPIO_PORT_PIN_MASK_FROM_DT_NODE(node_id)		\
-	GPIO_PORT_PIN_MASK_FROM_NGPIOS(DT_PROP(node_id, ngpios))
+	GPIO_DT_PORT_PIN_MASK_EXC(node_id)
```

Comment:
This change, and the change below are the only ones that need to be made in this commit.
I wouldn't even deprecate the macro GPIO_PORT_PIN_MASK_FROM_NGPIOS() or make any changes to the drivers.

Reply:
I would not agree. GPIO_PORT_PIN_MASK_FROM_NGPIOS probably is not deprecated itself, but the way how it's using definitely is : 

* drivers/gpio/gpio_ite_it8xxx2.c and drivers/gpio/gpio_kscan_ite_it8xxx2.c IMHO were suddenly skipped by sed script from        4e553f31fa43efe0272f2e26935e2cbaf4f0c1fb commit,
* drivers/gpio/gpio_stm32.c and drivers/gpio/gpio_npcx.c are using the gpio_controller YAML, so should handle the `gpio-reserved-ranges` too
 *  only drivers/gpio/gpio xlnx axi.c has non-standard syntax, so @henrikbrixandersen should makes the decision.
 
Btw it's exactly why I resisted to make this modification as a commit, but a PR :). Let's flame go on...

Reply:
> I would not agree. GPIO_PORT_PIN_MASK_FROM_NGPIOS probably is not deprecated itself, but the way how it's using definitely is :
> 
>     * drivers/gpio/gpio_ite_it8xxx2.c and drivers/gpio/gpio_kscan_ite_it8xxx2.c IMHO were suddenly skipped by sed script from        [4e553f3](https://github.com/zephyrproject-rtos/zephyr/commit/4e553f31fa43efe0272f2e26935e2cbaf4f0c1fb) commit,
> 
>     * drivers/gpio/gpio_stm32.c and drivers/gpio/gpio_npcx.c are using the gpio_controller YAML, so should handle the `gpio-reserved-ranges` too
> 
>     * only drivers/gpio/gpio xlnx axi.c has non-standard syntax, so @henrikbrixandersen should makes the decision.
> 
> 
> Btw it's exactly why I resisted to make this modification as a commit, but a PR :). Let's flame go on...

You can create an issue to track drivers that need to be updated.
I believe I made my suggested changes clear.

Reply:
Hmm, sound reasonable for me. Ok. I'll revert deprecated mark.

Reply:
Since, unfortunately, the standard Zephyr's create issue->create PR->merge procedure will delay the adoption of changes indefinitely, I'm returning the changes of the  drivers.
At least this PR has a chance of being accepted for the foreseeable future.